lib/ansible/module_utils/tacp.py

```diff
@@ -347,3 +347,52 @@ def get_resource_by_uuid(uuid, component, api_client):
     #                 uuid=datacenter.uuid, uuid)
     #         except ApiException as e:
     #             return "Exception when calling get_firewall_profiles_using_get"
+
+
+class ApplicationGroupResource(object):
+    def __init__(self, api_client, name, datacenter_uuid=None):
+        self.api = tacp.ApplicationGroupsApi(api_client)
+        self.name = name
+        self.datacenter_uuid = datacenter_uuid
+
+    def get(self):
```

Comment:
I would go a different approach. Make `ApplicationGroupResource` a model with `datacenter_uuid`, `name`, and `uuid` as fields. Then `get` and `get_or_create` can be staticmethods in which you pass in `name` and `datacenter_uuid` and it returns a proper `ApplicationGroupResource` instance instead of an unstructured dictionary.

Reply:
I wanted to get rid of any method arguments for that case where we deal with a specific resource. But I see what you mean and it might be a good idea for the future, where the code might get a bit more complicated. Until then I would like to avoid any premature optimizations.